firmware-support/bittide-sys/src/uart.rs

```diff
@@ -76,9 +84,9 @@ impl Uart {
 
     /// The `try_send` function attempts to send the given data to the UART. If
     /// the UART is unable to accept the data, it returns an error.
-    pub fn try_send(&mut self, data: u8) -> Result<(), ()> {
-        if self.read_status().tx_full {
-            Err(())
+    pub fn try_send(&mut self, data: u8) -> Result<(), TransmitBufferFull> {
+        if self.read_status().transmit_buffer_full {
+            Err(TransmitBufferFull)
         } else {
             unsafe {
                 self.payload_addr.write_volatile(data);
```

Comment:
I like the names better, but yeah, this feels like it should go in a separate commit.

Reply:
I did that right here because those were things clippy was complaining about for a while :see_no_evil: Probably should have been a different commit, yeah

Reply:
Split into a separate commit!